core/src/main/java/io/questdb/griffin/model/QueryModel.java

```diff
@@ -302,19 +297,34 @@ public void addUpdateTableColumnMetadata(int columnType, String columnName) {
         updateTableColumnNames.add(columnName);
     }
 
-    /* Determines whether this model allows pushing columns from parent model(s).
-     * If this is a UNION, EXCEPT or INTERSECT or contains a select distinct then it can't be done safely. */
+    /**
+     * Determines whether this model allows pushing columns from parent model(s).
+     * If this is a UNION, EXCEPT or INTERSECT or contains a SELECT DISTINCT then it can't be done safely.
+     */
     public boolean allowsColumnsChange() {
         QueryModel union = this;
         while (union != null) {
-            if (union.getSetOperationType() != QueryModel.SET_OPERATION_UNION_ALL ||
-                    union.getSelectModelType() == QueryModel.SELECT_MODEL_DISTINCT) {
+            if (union.getSetOperationType() != QueryModel.SET_OPERATION_UNION_ALL
+                    || union.getSelectModelType() == QueryModel.SELECT_MODEL_DISTINCT) {
                 return false;
             }
-
             union = union.getUnionModel();
         }
+        return true;
+    }
 
+    /**
+     * Determines whether this model allows pushing columns to nested models.
+     * If this is a SELECT DISTINCT then we should the parent model will contain the necessary columns.
```

Comment:
typo ?

Reply:
A typo indeed. Fixed in 8d841ae0c1328e2b18583fff5b03a79d91e33fe5